extension/chrome/elements/pgp_pubkey.ts

```diff
@@ -67,18 +67,24 @@ Catch.try(async () => {
       if (!isUsableButExpired && ! await pubs[0].getEncryptionKey() && ! await pubs[0].getSigningKey()) {
         showKeyNotUsableError();
       } else {
+        if (compact) {
+          $('.hide_if_compact_and_not_error').remove();
+        }
+        let emailText = '';
         if (pubs.length === 1) {
           const email = pubs[0].users[0].userId ? Str.parseEmail(pubs[0].users[0].userId ? pubs[0].users[0].userId!.userid : '').email : undefined;
           if (email) {
+            emailText = email;
             $('.input_email').val(email); // checked above
-            $('.email').text(email);
           }
         } else {
-          $('.email').text('more than one person');
+          emailText = 'more than one person';
           $('.input_email').css({ display: 'none' });
           const pubToEmail = (pubkey: OpenPGP.key.Key) => Str.parseEmail(pubkey.users[0].userId ? pubkey.users[0].userId!.userid : '').email;
           Xss.sanitizeAppend('.add_contact', Xss.escape(' for ' + pubs.map(pubToEmail).filter(e => !!e).join(', ')));
         }
+        Xss.sanitizePrepend('#pgp_block.pgp_pubkey .result', `<span>This message includes a Public Key for <span class= "email">${emailText}</span>.</span>`);
```

Comment:
Better, but please don't fully rely on sanitizing only, because it may also fail. If it's a simple case like this, it is appropriate to xss escape the email before running it through sanitization.

Security is always applied in layers, like an onion. If one layers fails, which it may anytime, there should be another layer that ends up saving your (user's) butt.

Reply:
Okay, Got It,
I will sanitize any text that we get from the user :)